### ISSUE TITLE
Broken build when using old libzmq, and wrong docs

### ISSUE THREAD

**Issue by kanzure** [bitcoin/bitcoin#6734]
Building with libzmq-dev from debian (so libzmq 2.x) results in the following error about an undefined symbol `zmq_ctx_destroy`:

```
make[3]: Leaving directory '/home/kanzure/local/bitcoin/bitcoin/src/secp256k1'
  CXX      zmq/libbitcoin_zmq_a-zmqabstractnotifier.o
  CXX      zmq/libbitcoin_zmq_a-zmqnotificationinterface.o
zmq/zmqnotificationinterface.cpp: In member function ‘void CZMQNotificationInterface::Shutdown()’:
zmq/zmqnotificationinterface.cpp:117:33: error: ‘zmq_ctx_destroy’ was not declared in this scope
         zmq_ctx_destroy(pcontext);
                                 ^
Makefile:4328: recipe for target 'zmq/libbitcoin_zmq_a-zmqnotificationinterface.o' failed
make[2]: *** [zmq/libbitcoin_zmq_a-zmqnotificationinterface.o] Error 1
```

Switching to libzmq3-dev results in a successful build for me.

At `doc/zmq.md` there is text saying:

> The ZeroMQ feature in Bitcoin Core uses only a very small part of the ZeroMQ C API, and is thus compatible with any version of ZeroMQ from 2.1 onward, including all versions in the 3.x and 4.x release series. Typically, it is packaged by distributions as something like libzmq-dev.

But see http://zeromq.org/intro:get-the-software which marks even libzmq 3.x as "legacy"; they follow semver and have major incompatibilities between major version field changes. Recent versions of libzmq should be compiled from source from [here](http://zeromq.org/intro:get-the-software) instead of APT repos.

Also, you can't have sockets from 2.x and 3.x talk to 4.x anyway.

**Comment by luke-jr** [comment]
Minor note: Sockets in 4.0 will talk with 4.1, but there are apparently subtle incompatibilities.

**Comment by jonasschnelli** [comment]
I can confirm this issue.
We should update `configure.ac` that it will only accept `libzmq3` in cases of `--enable-zmq`.

**Comment by jmcorgan** [comment]
I should have a PR with an update for configure.ac, the docs, and release notes soon.

**Comment by jmcorgan** [comment]
See #6736

**Comment by laanwj** [comment]
Closed by #6736
